firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Driver/Serial/Backend/frames.txt

```diff
@@ -0,0 +1,138 @@
+   THEN("The input buffer is unchanged after transform")
+
+  WHEN("transform is called on an input buffer that contains these bytes '\x11\x22\x00\x33'")
+   THEN("The transform method reports ok status")
+   THEN("The output buffer is as expected '\x03\x11\x22\x02\x33'")
+   THEN("The input buffer is unchanged after transform")
+
+  WHEN("transform is called on an input buffer that contains these bytes '\x11\x22\x33\x44'")
+   THEN("The transform method reports ok status")
+   THEN("The output buffer is as expected '\x05\x11\x22\x33\x44'")
+   THEN("The input buffer is unchanged after transform")
+
+  WHEN("transform is called on an input buffer that contains the char 'x'")
+   THEN("The transform method reports ok status")
+   THEN("The output buffer is as expected '\x02\x78'")
+   THEN("The input buffer is unchanged after transform")
+
+  WHEN("transform is called on an input buffer that contains the char 'xy'")
+   THEN("The transform method reports ok status")
+   THEN("The output buffer is as expected '\x03\x78\x79'")
+   THEN("The input buffer is unchanged after transform")
+
+  WHEN("transform is called on an input buffer containing the string 'Hello world')
+   THEN("The transform method reports ok status")
+   THEN("The output buffer is as expected '\x0cHello World'")
+   THEN("The input buffer is unchanged after transform")
+
+  WHEN("transform is called on an input buffer that contains a 16-byte sensor measurements message payload")
+   THEN("The transform method reports ok status")
+   THEN("The output buffer has an expected sequence of 17 bytes")
+   THEN("The input buffer is unchanged after transform")
+
+Scenario: The FrameReceiver class correctly generates COBS decoded bodies from encoded buffers
+
+ GIVEN("A FrameReceiver object constructed with default parameters")
+ // assertions carried out in given:
+ // output_buffer is empty
+
+  WHEN("255 non-delimiter frame body is passed as input and output is called after that")
+   THEN("The input method reports ok status for all 255 non delimiter-bytes")
+   THEN("The output method reports waiting status")
+   THEN("The output buffer remains empty")
+
+  WHEN("256 non-delimiter frame body is passed as input and output is called after that")
+   THEN("The input method reports ok status for all 255 non-delimiter bytes and invalid_length for the 256th byte")
+   THEN("The output method reports invalid_length status")
+   THEN("The output buffer remains empty")
+
+  WHEN("128 non-delimiter frame body followed by a delimiter are passed as input, and output is called after that")
+   THEN("The input method reports ok status for all 128 non-delimiter bytes and output_ready for 129th byte")
+   THEN("The output method reports ok status")
+   THEN("The output buffer has an expected sequence of 128 bytes")
+
+  WHEN("256-byte frame body with a delimiter at the 129th and 256th-byte position is passed as input, and output is called after that")
```

Comment:
WHEN 256 bytes are passed as input, where the input's delimiter bytes are at positions 129 and 256, and then output is called afterwards